packages/components/tsup.config.ts

```diff
@@ -1,27 +1,65 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
 import svgr from 'esbuild-plugin-svgr';
 import { defineConfig, Options } from 'tsup';
-import tsconfig from './tsconfig.json';
 
-const options = defineConfig({
+const options: Options = {
   format: 'esm',
-  target: tsconfig.compilerOptions.target as Options['target'],
-  clean: true,
+  target: 'es2021',
   dts: true,
-});
+  outExtension: () => ({ js: '.js' }),
+};
+
+const PATH_WITH_SVG = [
+  'src/logos/index.tsx',
+  'src/components/icons/index.ts',
+  'src/components/decorations/index.tsx',
+];
 
 export default defineConfig([
   {
+    ...options,
     name: 'components',
-    entry: {
-      index: 'src/index.ts',
-      products: 'src/products.tsx',
-      logos: 'src/logos/index.tsx',
-    },
+    entry: [
+      'src/**/*.{ts,tsx}',
+      'src/static/illustrations/*.png',
+      '!**/*.stories.{ts,tsx}',
+      '!src/helpers/dummy.ts',
+      '!src/*.d.ts',
+      ...PATH_WITH_SVG.map(filePath => `!${filePath}`),
+    ],
+    bundle: false,
     loader: {
       '.png': 'copy',
     },
-    outExtension: () => ({ js: '.js' }),
-    external: ['semver'],
+    async onSuccess() {
+      const serverPackageJSON = path.join(process.cwd(), 'dist', 'server', 'package.json');
+      await fs.writeFile(serverPackageJSON, '{"type":"module","sideEffects":false}');
+    },
+    plugins: [
+      {
+        // Strip `node:` prefix from imports because
+        // Next.js only polyfills `path` and not `node:path` for browser
+        name: 'strip-node-colon',
+        renderChunk(code) {
+          const replaced = code.replaceAll(/ from "node:(?<moduleName>.*?)";/g, matched =>
+            matched.replace('node:', ''),
+          );
+          return { code: replaced };
+        },
+      },
+    ],
+  },
+  // We need to bundle svg files separately, due required `bundle: true` option which is set by default
```

Comment:
with `bundle: true` default behaviour

<img width="1138" alt="image" src="https://github.com/user-attachments/assets/c7139ba9-ea1c-4070-9317-b088525bf742">

with `bundle: false` (used in 1st `defineConfig` option)
<img width="989" alt="image" src="https://github.com/user-attachments/assets/b7f463a5-9786-45dc-8d5f-cbabae536362">

Maybe you have an idea how we can fix it without declaring 2nd option with `bundle: true` ? 😅

Reply:
Oh, okay. So we need `bundle: true` otherwise the SVGs won't be transformed.

And for other cases, we leave the bundling to the application to allow it to code split better and make the debugging easier.

That makes a ton of sense, thank you.